### PR TITLE
Doc for the viscoplastic theory

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR is for adding a doc describing the viscoelastic theory behind the PR #2004. It belongs the tasks listed in the EPIC #2401

### CODE REVIEW

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   \hat{\boldsymbol{\sigma}}^{t+\Delta t} = \boldsymbol{\sigma}^t + \tensor{c}^{t+\Delta t}:\Delta \boldsymbol{\epsilon}^{t+\Delta t}
+
+The tangent stiffness tensor is updated using the following equivalent approximation:
+
+.. math::
+   \tensor{c}^{t+\Delta t} = r_t \tensor{c}^e + (1-r_t) \tensor{c}^{t}
+
+Here, :math:`\tensor{c}^e` is the elastic stiffness tensor.
+
+The name of the viscoplastic solver is defined by adding the prefix `Visco` to the name of the elasto-plastic solver used to compute the inviscid stress :math:`\overline{\boldsymbol{\sigma}}`. For example, the solver `Visco Drucker-Prager` corresponds to cases where the inviscid stress is computed by the `Drucker-Prager` solver. It is interesting to note that equivalent viscoelastic solutions can also be obtained using the Duvault-Lions algorithm by updating the inviscid stress with an elastic solver.
```
Comment (jhuang2601) [inline src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:44]:
```suggestion
The name of the viscoplastic solver is formed by adding the prefix `Visco` to the name of the elasto-plastic solver used to compute the inviscid stress :math:`\overline{\boldsymbol{\sigma}}`. For example, the solver `Visco Drucker-Prager` corresponds to cases where the inviscid stress is computed by the `Drucker-Prager` solver. It is interesting to note that equivalent viscoelastic solutions can also be obtained using the Duvault-Lions algorithm by updating the inviscid stress with an elastic solver.
```

**Review by jhuang2601 — APPROVED** [review]
Thanks for preparing this doc.

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
+.. _ViscoPlasticModel:
+
+############################################
+Vico-Plastic Models
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:4]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+############################################
+
+The classical `Perzyna-type viscoplasticity models <https://www.sciencedirect.com/science/article/abs/pii/S0065215608700097>`__ are not suitable for nonsmooth multisurface rate-dependent plasticity models because of the unclearly defined nested viscoplastic loading surfaces. An alternative formulation, the `Duvaut-Lions viscoplastic theory <https://onlinelibrary.wiley.com/doi/abs/10.1002/nme.1620261003>`__, precludes these difficulties by not using the concept of nested viscoplastic loading surfaces. The viscoplastic constitutive equation that relates the stress :math:`\boldsymbol{\sigma}` and the viscoplastic strain rate :math:`\dot{\boldsymbol{\epsilon}^{vp}}` is given by:
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:7]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
+   \boldsymbol{\sigma} - \bar{\boldsymbol{\sigma}} = \frac{1}{t_*}\tensor{c}:\dot{\boldsymbol{\epsilon}^{vp}}
+
+Here, :math:`\bar{\boldsymbol{\sigma}}` represents the inviscid stress, which is the rate-independent elasto-plastic stress part that can be obtained by using elasto-plastic solvers such as Drucker-Prager, CamClay, etc. :math:`\tensor{c}` is the tangent stiffness tensor and :math:`t_*` is the relaxation time parameter, which is measured in units of time. The viscoplastic strain rate :math:`\dot{\boldsymbol{\epsilon}}^{vp}` can be approximated using the following finite difference:
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:12]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
+   \dot{\boldsymbol{\epsilon}^{vp}} = \frac{1}{\Delta t}(\Delta \boldsymbol{\epsilon} - \Delta \boldsymbol{\epsilon}^{elas})
+
+Here, :math:`\Delta t` is the time increment, :math:`\Delta \boldsymbol{\epsilon}` is the total strain increment, and :math:`\Delta \boldsymbol{\epsilon}^{elas}` is the elastic part of the strain increment. It is important to note that the elastic strain increment is related to the stress increment through Hook's elastic law.
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:17]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
+   \Delta \boldsymbol{\sigma} = \tensor{c}:\Delta \boldsymbol{\epsilon}^{elas}
+
+With some arrangements, we can obtain the following formula to update the stress tensor of the Duvaut-Lions elasto-viscoplastic material:
+
+.. math::
+   \boldsymbol{\sigma} = r_t \boldsymbol{\sigma}^{trial} + (1-r_t) \bar{\boldsymbol{\sigma}}
+
+Here, the time ratio :math:`r_t` is calculated from the relaxation time parameter :math:`t_*` and the time increment :math:`\Delta t` as:
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:27]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
+   r_t = \frac{1}{1+\Delta t/t_*}
+
+The trial stress tensor is computed using the strain increment, assuming elastic behavior:
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:32]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+Here, :math:`\Delta t` is the time increment, :math:`\Delta \boldsymbol{\epsilon}` is the total strain increment, and :math:`\Delta \boldsymbol{\epsilon}^{elas}` is the elastic part of the strain increment. It is important to note that the elastic strain increment is related to the stress increment through Hook's elastic law.
+
+.. math::
+   \Delta \boldsymbol{\sigma} = \tensor{c}:\Delta \boldsymbol{\epsilon}^{elas}
+
+With some arrangements, we can obtain the following formula to update the stress tensor of the Duvaut-Lions elasto-viscoplastic material:
+
+.. math::
+   \boldsymbol{\sigma} = r_t \boldsymbol{\sigma}^{trial} + (1-r_t) \bar{\boldsymbol{\sigma}}
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:25]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst (hunk 1 above)
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:44]: Done

File: src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
+   \dot{\boldsymbol{\epsilon}^{vp}} = \frac{1}{\Delta t}(\Delta \boldsymbol{\epsilon} - \Delta \boldsymbol{\epsilon}^{elas})
+
+Here, :math:`\Delta t` is the time increment, :math:`\Delta \boldsymbol{\epsilon}` is the total strain increment, and :math:`\Delta \boldsymbol{\epsilon}^{elas}` is the elastic part of the strain increment. It is important to note that the elastic strain increment is related to the stress increment through Hook's elastic law.
+
+.. math::
+   \Delta \boldsymbol{\sigma} = \tensor{c}:\Delta \boldsymbol{\epsilon}^{elas}
+
+With some arrangements, we can obtain the following formula to update the stress tensor of the Duvaut-Lions elasto-viscoplastic material:
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/coreComponents/constitutive/docs/solid/ViscoPlasticity.rst:22]: Done

**Review by herve-gross — APPROVED** [review]

**Pull request merged**